pyproject.toml

```diff
@@ -165,13 +169,29 @@ enable_error_code = ["ignore-without-code", "redundant-expr", "truthy-bool"]
 warn_unreachable = true
 disallow_untyped_defs = false
 disallow_incomplete_defs = false
+# data-apis/array-api#589
+disallow_any_expr = false
 
 [[tool.mypy.overrides]]
 module = "array_api_extra.*"
 disallow_untyped_defs = true
 disallow_incomplete_defs = true
 
 
+# pyright
+
+[tool.pyright]
+include = ["src", "tests"]
+pythonVersion = "3.10"
+pythonPlatform = "All"
+typeCheckingMode = "strict"
```

Comment:
You could also try setting this to `"recommended"` or `"all"`, which unleases the full power of basedpyright.
See https://docs.basedpyright.com/latest/configuration/config-files/#diagnostic-settings-defaults for what this exactly enables

Reply:
Switching to `all` introduces one error:

`array-api-extra/tests/test_funcs.py:8:8 - error: Stub file not found for "array_api_strict" (reportMissingTypeStubs)`